src/overlays/actors/ovl_En_Arrow/z_en_arrow.c

```diff
@@ -33,45 +34,710 @@ const ActorInit En_Arrow_InitVars = {
+                this->unk_264->world.pos.y = ((sp54.y <= sp9C.y) ? 1.0f : -1.0f) + sp9C.y;
+                this->unk_264->world.pos.z = ((sp54.z <= sp9C.z) ? 1.0f : -1.0f) + sp9C.z;
+
+                Math_Vec3f_Diff(&this->unk_264->world.pos, &this->actor.world.pos, &this->unk_268);
+                this->unk_264->flags &= ~0x8000;
+                this->unk_264 = NULL;
+            } else {
+                Math_Vec3f_Sum(&this->actor.world.pos, &this->unk_268, &this->unk_264->world.pos);
+            }
+
+            if ((this->unk_262 != 0) && (this->unk_264 != NULL)) {
+                this->unk_264->flags &= ~0x8000;
+                this->unk_264 = NULL;
+            }
+        } else {
+            this->unk_264 = NULL;
+        }
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Arrow/func_8088ACE0.s")
+#endif
+
+void func_8088B630(EnArrow* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->arrow.skelAnime);
+
+    if (this->actor.wallBgId != BG_ACTOR_MAX) {
+        BgCheck2_UpdateActorAttachedToMesh(&globalCtx->colCtx, this->actor.wallBgId, &this->actor);
+    }
+
+    if (DECR(this->unk_260) == 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void func_8088B6B0(EnArrow* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->arrow.skelAnime);
+    Actor_MoveWithGravity(&this->actor);
+    func_8088AA98(this, globalCtx);
+
+    if (DECR(this->unk_260) == 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void EnArrow_Update(Actor* thisx, GlobalContext* globalCtx) {
+    static Vec3f D_8088C2CC = { 0.0f, 0.5f, 0.0f };
+    static Vec3f D_8088C2D8 = { 0.0f, 0.5f, 0.0f };
+    static Color_RGBA8 D_8088C2E4 = { 255, 255, 100, 255 };
+    static Color_RGBA8 D_8088C2E8 = { 255, 50, 0, 0 };
+    s32 pad;
+    EnArrow* this = THIS;
+    Player* player = GET_PLAYER(globalCtx);
+
+    if ((this->unk_263 != 0) ||
+        ((this->actor.params >= ENARROW_0) && ((this->actor.params == ENARROW_7) || (player->unk_D57 != 0))) ||
+        !func_80123358(globalCtx, player)) {
+        this->actionFunc(this, globalCtx);
+    }
+
+    if ((this->actor.params >= ENARROW_3) && (this->actor.params < ENARROW_6)) {
+        s16 sp44[] = { ACTOR_ARROW_FIRE, ACTOR_ARROW_ICE, ACTOR_ARROW_LIGHT };
+
+        if (this->actor.child == NULL) {
+            Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, sp44[this->actor.params - ENARROW_3],
+                               this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, 0);
+        }
+    } else if (this->actor.params == ENARROW_0) {
+        func_800B0EB0(globalCtx, &this->unk_234, &D_8088C2CC, &D_8088C2D8, &D_8088C2E4, &D_8088C2E8, 100, 0, 8);
+    }
+    Math_Vec3f_Copy(&this->actor.home.pos, &this->actor.prevPos);
+}
+
+typedef struct {
+    /* 0x00 */ Vec3f unk_00;
+    /* 0x0C */ Vec3f unk_0C;
+} EnArrowVecs;
```

Comment:
Are you sure this is a real struct and not a `Vec3f[2]`?

Reply:
Pretty sure it's some kinda of struct. A Vec3f[2] doesn't solve the problem with ```
func_8088B88C(globalCtx, this, (EnArrowVecs*)&D_8088C2EC[3].unk_0C);```

Reply:
Hmmmm. That part looks a bit like a fake match because that doesn't make much sense

Reply:
I know, but I can't figure out what it's supposed to be.

Reply:
Could you add a comment saying it can be a fake match? That way is a bit easier to look for those in the future

Reply:
As discussed in discord, the correct struct would probably be 
```c
typedef struct EnArrowUnkStruct {
    /* 0x00 */ Vec3f unk_00[2];
    /* 0x18 */ Vec3f unk_18[2][2];
    /* 0x48 */ Vec3f unk_48;
} EnArrowUnkStruct; // size = 0x54
```